src/components/panel/panel.js

```diff
@@ -0,0 +1,284 @@
+ *           '    </md-list-item>' +
+ *           '  </md-list>' +
+ *           '  <div class="actions">' +
+ *           '    <md-button ng-click="ctrl.closeDialog()">CANCEL</md-button>' +
+ *           '  </div>' +
+ *           '</div>',
+ *         clickOutsideToClose: true,
+ *         escapeToClose: true,
+ *         focusOnOpen: true
+ *       }
+ *       panelRef = $mdPanel.create(config);
+ *       panelRef.open()
+ *           .finally(function() {
+ *             panelRef = undefined;
+ *           });
+ *     }
+ *   }
+ *
+ *   function DialogController($mdPanelRef, toppings) {
+ *     var toppings;
+ *     var favoriteTopping;
+ *
+ *     function closeDialog() {
+ *       $mdPanelRef.close();
+ *     }
+ *   }
+ * })(angular);
+ * </hljs>
+ */
+
+/**
+ * @ngdoc method
+ * @name $mdPanel#create
+ * @description
+ * Creates a panel with the specified options.
+ *
+ * @param opt_config {Object=} Specific configuration object that may contain the
+ * following properties:
+ *
+ *   N/A
+ *
+ * TODO(ErinCoughlan): Add the following config options.
+ *   - `groupName` - `{string=}`: Name of panel groups. This group name is used for
+ *     configuring the number of open panels and identifying specific behaviors for
+ *     groups. For instance, all tooltips will be identified using the same groupName.
+ *   - `maxOpen` - `{number=}` : Maximum number of panel in a specific group that
+ *     can be opened at a given time. If multiple panels have the same groupName, but
+ *     different values of maxOpen, only the first one will be considered. A warning
+ *     will be logged in the console.
+ *   - `template` - `{string=}`: HTML template to show in the dialog. This **must** be
+ *     trusted HTML with respect to Angular’s
+ *     [$sce service](https://docs.angularjs.org/api/ng/service/$sce).
+ *   - `templateUrl` - `{string=}`: The URL that will be used as the content of
+ *     the panel.
+ *   - `controller` - `{(function|string)=}`: The controller to associate with the
+ *     panel. The controller will be injected with a reference to the returned
+ *     panelRef, which allows the panel to be closed, hidden, and shown.
+ *     Any fields passed in through locals or resolve will be bound to the controller.
+ *   - `controllerAs` - `{string=}`: An alias to assign the controller to on the
+ *     scope.
+ *   - `locals` - `{Object=}`: An object containing key/value pairs. The keys will be
+ *     used as names of values to inject into the controller. For example,
+ *     `locals: {three: 3}` would inject `three` into the controller, with the value
+ *     3.
+ *   - `resolve` - `{Object=}`: Similar to locals, except it takes promises as
+ *     values. The panel will not open until all of the promises resolve.
+ *   - `parent` - `{Element=}`: The element to append the dialog to. Defaults to
+ *     appending to the root element of the application.
+ *   - `position` - `{(string|Element|Object)=}`: The query selector, DOM element,
```

Comment:
What do you think about this API shape instead:
```js
postion: $mdPanelPosition.relativeTo(someElement).offsetX(-20).offsetY(50);
// or
postion: $mdPanelPosition.absolute().top('50%').left('50%');
```
The end result is that `$mdPanel` will always get a position of a certain shape rather than having to interpret everything.